quantinuum_qircheck/qircheck.py

```diff
@@ -22,6 +22,38 @@
 import pyqir as pq
 
 
+class _cycle_check:
+    # checks for cycles in the CFG
+
+    def __init__(self) -> None:
+        # gray nodes/blocks
+        # marks all blocks that are currently traversed
+        self.current_blocks: set[pq.BasicBlock] = set()
+
+        # black nodes/blocks
+        # marks all blocks that have been traversed
+        self.visited_blocks: set[pq.BasicBlock] = set()
+
+    def check_for_cycles(self, block: pq.BasicBlock) -> None:
+        # checks if the given block is part of a cycle in the CFG
```

Comment:
The description of this method ("checks if the given block is part of a cycle in the CFG") is incomplete since it doesn't mention the dependence on `current_blocks` and `visited_blocks`. Could you expand it?

Reply:
Done, see [a21e218](https://github.com/CQCL/quantinuum-qircheck/pull/8/commits/a21e218ff3d29dfbc7e8a0fd4d9e698fc5b23c55)